deepfence_server/pkg/integration/splunk/splunk.go

```diff
@@ -23,64 +31,93 @@ func New(ctx context.Context, b []byte) (Splunk, error) {
 }
 
 func (s Splunk) SendNotification(ctx context.Context, message string, extras map[string]interface{}) error {
+	s.client = utils.GetInsecureHttpClient()
 	var msg []map[string]interface{}
 	d := json.NewDecoder(strings.NewReader(message))
 	if err := d.Decode(&msg); err != nil {
 		fmt.Println("Failed to unmarshal message for splunk", err)
 		return err
 	}
+
+	numRoutines := (len(msg) / 10)
+	if numRoutines == 0 {
+		numRoutines = 1
+	} else if numRoutines > 10 {
+		numRoutines = 10
+	}
+
+	senderChan := make(chan []byte, 500)
+	var wg sync.WaitGroup
+	for i := 0; i < numRoutines; i++ {
+		wg.Add(1)
+		go s.Sender(senderChan, &wg)
```

Comment:
How much performance gain are we getting with those go routines?

Reply:
@noboruma We will not be flooding the remote servers as we can have maximum of 40 connections at any given time. 
Reason being we have max of 4 routines for an integration(each for vulnerability, secret, malware, compliance). For each routine, we have max of 10 go routines. That's one of the reason for limiting the routines to 10.

As for the performance, please find below the numbers (time is in milliseconds)
**Base Case**: Logs when the number of routines = 1 
Mon, 11 Sep 2023 04:18:13 +0000 INF splunk/splunk.go:61 SPLUNK::SendNotification complete, numRoutines:1
Mon, 11 Sep 2023 04:18:13 +0000 INF cronjobs/notification.go:221 Notification sent Vulnerability scan 87 messages using splunk id 3, time taken:21644

**With Changes** : Logs when the number of routines = 8
Mon, 11 Sep 2023 04:27:48 +0000 INF splunk/splunk.go:59 SPLUNK::SendNotification complete, numRoutines:8
Mon, 11 Sep 2023 04:27:48 +0000 INF cronjobs/notification.go:221 Notification sent Vulnerability scan 87 messages using splunk id 3, time taken:3418

From the numbers, we can see the speedup of around 6 times.

Reply:
@varunsharma0286 Can you check you are not getting 429 error in all integrations with this change?
If no 429 error, we are good.

Reply:
@ramanan-ravi No 429 :)
Also, these changes are specifically for "Splunk" and "Teams"

Reply:
@varunsharma0286 sounds good, one last verification, are those logs generated with the same code but only change is `numRoutines = 1`?

Reply:
Yups... that's correct.

Reply:
Not for this PR but we would benefit having a pool across all the integrations, simplifying the code and making it easier to maintain